---
title: "Genome-Wide Association Study of Brain Arteriolosclerosis"
author:
  - Lincoln M.P. Shade^1^
  - Yuriko Katsumata^1,2^
  - Timothy J. Hohman^3^
  - Julie A. Schneider^4^
  - Kwangsik Nho
  - Andrew Saykin
  - Shubhabrata Mukherjee
  - Alzheimer’s Disease Genetics Consortium
  - Peter T. Nelson^2,5^
  - David W. Fardo^1,2^
date: "`r Sys.Date()`"
output: word_document
bibliography: references.bib
csl: springer_basic.csl
link-citations: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE, include = FALSE)
```

```{r load-opts-packages-functions}
source("code/00_load_options_packages_functions.R")
```

```{r load-individual-level-data}
load("data/nacc_adgc/nacc_adgc_unrelated.RData")
load("data/rosmap/rosmap_unrelated.RData")
load("data/nacc_adgc_rosmap/nacc_adgc_rosmap_unrelated.RData")
load("data/adni_npc/nacc_adgc_rosmap_adni_unrelated.RData")
```

```{r load-single-variant-analysis-results, results = FALSE}
nacc_adgc_results <- fread("output/nacc_adgc/nacc_adgc_unrelated.assoc.logistic")
nacc_adgc_results[, SNP := strip_alleles(SNP)]
rosmap_results <- fread("output/rosmap/rosmap_unrelated.assoc.logistic")
nacc_rosmap_mega_results <- fread("output/nacc_adgc_rosmap/nacc_adgc_rosmap_unrelated.assoc.logistic")
adni_mega_results <- fread("output/adni_npc/nacc_adgc_rosmap_adni_unrelated.assoc.logistic")
act_meta_results <- fread("output/act/act__nacc_adgc_rosmap_adni_unrelated.meta")
```

```{r adni-mega-results, results=FALSE}
# adni_mega_top <- fread("output/adni_npc/nacc_adgc_rosmap_adni_unrelated.clumped")
# adni_mega_top <- adni_mega[SNP %in% adni_mega_top$SNP]
# setorder(adni_mega_top, P)
# adni_mega_top[, .(CHR, BP, SNP, P)] %>% kable() %>% kable_styling()
```

```{r act-meta-results, results=FALSE}
# setorder(act_meta, P)
# act_meta[P < 1e-5]
```

# Author Affiliations

^1^Department of Biostatistics, College of Public Health, University of Kentucky, Lexington, KY

^2^Sanders-Brown Center on Aging and Alzheimer's Disease Research Center, University of Kentucky, Lexington, KY

^3^Vanderbilt Memory & Alzheimer's Center, Department of Neurology, Vanderbilt University Medical Center, Nashville, Tennessee

^4^Departments of Neurology and Pathology, Rush University Medical Center, Chicago, IL, 60612, USA.

^5^Pathology and Laboratory Medicine, University of Kentucky, Lexington, KY

## Corresponding Author

Lincoln Shade

Email: [lincoln.shade\@uky.edu](mailto:lincoln.shade@uky.edu)

Phone: 859-893-5798

\newpage

# Abstract

Brain arteriolosclerosis (B-ASC) is a small-vessel cerebrovascular disease involving sclerotic thickening of arterioles in the brain. B-ASC pathology is common in aged autopsy cohorts, is associated with worse cognitive functioning, and is associated with other neuropathologies. The genetics of other contributors of dementia have been studied extensively, but to date no in-depth study has been conducted on genetic risk of autopsy-proven B-ASC. We perform the first genome-wide association study (GWAS) on B-ASC using multiple cohorts. We then follow-up on identified risk variants with functional analysis to investigate potential biological mechanisms. Individual GWAS and mega-analyses were conducted using data from participants in the National Alzheimer's Coordinating Center (NACC) and the Religious Orders Study and Memory and Aging Project (ROSMAP). Variants identified in GWAS were then checked for quantitative trait loci (QTL) associations in the Genotype Tissue Expression (GTEx) project. Colocalization analysis was then performed on identified QTL in brain and vascular tissues. Top variants were also examined for association with neuroimaging correlates of B-ASC in the Alzheimer's Disease Neuroimaging Initiative. One locus on chromosome six with lead variant rs2603462 is significantly associated with B-ASC in NACC and colocalizes with *ELOVL4* gene expression in the brain. Other variants in both cohorts achieved less stringent thresholds of association and colocalize with multiple QTL in GTEx. One locus on chromosome ten near *SORCS3* with lead variant rs7902929 suggestively associated with B-ASC in NACC was validated in ROSMAP. Genetic loci associated with B-ASC pathology were identified using multiple cohorts. Most loci associated with B-ASC in one cohort were not associated in the other. Multiple identified risk loci colocalize with gene expression or splicing QTL, providing evidence for potential biological mechanisms through which variants may influence disease risk. Due to the relatively small sample size and heterogeneity between the cohorts used, future studies should be done to better characterize genetic risk of B-ASC.

# Keywords

> Please provide 4 to 6 keywords which can be used for indexing purposes.

\newpage

# Introduction

Brain arteriolosclerosis (B-ASC) is a subtype of cerebral small vessel neuropathologic change characterized by thickening of arteriole walls in the brain. These changes can include hypertrophy or atrophy of vascular smooth muscle and luminal extracellular deposition of elastin and collagen [@ighodaro2017; @grinberg2010; @buchman2013]. B-ASC is commonly found in autopsied elderly individuals, with 39-80% of participants showing some B-ASC pathology in large autopsy studies [@ighodaro2017; @buchman2013; @chou2013 ]. B-ASC is associated with multiple neuropathologies including Alzheimer's disease (AD), limbic-predominant age-related TDP-43 encephalopathy (LATE), micro-infarcts, and large-vessel infarcts [@ighodaro2017; @arvanitakis2016; @neltner2014 ]. B-ASC is also associated with cognitive decline after adjusting for the presence of other neuropathologies [@ighodaro2017; @arvanitakis2016]. Despite the clinical importance of B-ASC, its risk factors, other than age and sex, remain largely uncharacterized. Hypertension and diabetes are established clinical risk factors for arteriolosclerosis in the kidneys, but their association with B-ASC is inconsistent; in an autopsy based cohort stratified by age of death (\<80 years vs. ≥80 years), hypertension was significantly associated with B-ASC in the younger group only, and diabetes was not associated with B-ASC in either group [@ighodaro2017; @wu2005; @ikee2010; @cameron2006].

Genome-wide association studies (GWAS) have provided a powerful resource for investigating genomic risk of complex diseases through analysis of millions of common genetic variants with diseases of interest, and have to date identified tens-of-thousands of variants associated with disease [@buniello2019]. GWAS have successfully been used to identify genetic risk loci for stroke and for imaging-based phenotypes of cerebrovascular disease such as white matter hyperintensities (WMH) and brain infarcts; however, we found no published GWAS for autopsy-proven B-ASC to date [@traylor2016; @malik2018; @chauhan2019]. Given the unique and complex structures of brain arterioles and their associated structures, including astrocytes and other components of the blood-brain barrier, there is reason to suspect that the genomic risk for B-ASC may not be wholly shared with other cerebrovascular phenotypes. In one study of Religious Orders Study/Memory and Aging Project (ROSMAP) participants, 167 independent genetic variants previously meeting genome-wide significance threshold (P \< 5×10-8) in GWAS of stroke or stroke risk factorswere tested for association with B-ASC pathology [@chou2013]. The authors found 6 variants nominally associated with B-ASC at the 0.01 \< P-value \< 0.05 significance level, none of which remained significant after correction for multiple tests [@chou2013]. Investigating genomic risk factors of B-ASC at the genome-wide level may provide important insights into its pathophysiological development as well as its relationship to neuroimaging and other neuropathological phenotypes.

In the present study, we conduct a GWAS using B-ASC pathology as the endophenotype analyzing neuropathology and genotype data across four aged autopsy cohorts. In the first stage of our study, we analyzed data from the National Alzheimer's Coordinating Center (NACC) Neuropathology Dataset linked to genotype data from the Alzheimer's Disease Genetics Consortium (ADGC; NACC/ADGC when referring to combined datasets). In the second stage, we performed a GWAS using data from ROSMAP participants and then mega-analyze both datasets together. In the third stage, we attempted to replicate our findings froms stages one and two in two smaller autopsy cohorts consisting of Alzheimer's Disease Neuroimaging Inititative (ADNI) and Adult Changes in Thought (ACT) participants. To investigate potential biologically functional correlates to disease risk, we then perform colocalization analysis on B-ASC associated variants identified as quantitative trait loci (QTL) using data from the Genetic Tissue Expression (GTEx) Project and gene-based association analyses [@thegeno2013].

# Methods

## Study Participants

In our first study, we linked neuropathology data from participants in thirty-one National Institute on Aging-funded Alzheimer's Disease Research Centers (ADRCs) studies from the NACC Neuropathology Data Set to ADGC genotype data [@welcome; @besser2018]. Each ADRC has its own recruitment strategies, populations, and study design, and data are collected and aggregated by NACC. A total of **XXXX** participants had both neuropathology and genotype data available. We then excluded all participants diagnosed with any of nineteen rare neurological conditions (see **Supplementary Table S1** for full exclusion criteria).

The ROSMAP study has been previously described in detail and consists of harmonized data from two longitudinal cohorts: The Religious Orders Study (ROS) and the Memory and Aging Project (MAP). ROS began in 1994 and recruited older Catholic Priests and Sisters from around the United States. MAP began in 1997 and recruited older adults who at the time had no diagnosis of dementia. A total of **XXXX** ROSMAP participants had both autopsy and genotyping data available.

The ADNI study...

The ACT study...

## **Definitions of B-ASC variables used**

In the NACC Neuropathology Data Set, B-ASC is graded as an ordinal variable with possible values of 0 (none), 1 (mild), 2 (moderate), or 3 (severe). Grading is performed by trained Neuropathologists at each ADRC and is a global rating, meaning no instruction is given to examine specific brain regions for B-ASC. Both the ADNI and ACT studies follow the NACC Neuropathology Form when grading neuropathological phenotypes. The B-ASC variable in ROSMAP is graded on B-ASC histological changes exclusively in the basal ganglia. Vessel intimal pathology is first graded on a scale from 0 (none) to 6 (severe), and then collapsed to a four-level ordinal variable with the same labels as in NACC: 0 (none), 1 (mild), 2 (moderate), and 3 (severe) [@buchman2011].

## **Quality control (QC) of genotype data**

The data in our study were all subset from datasets whose QC measures have been previously described . We performed additional standard QC procedures on all genotyping data using PLINK v1.9 and KING [@chang2015; @chang; @purcell2007; @manichaikul2010; @marees2018]. Variants were excluded if they had (1) a minor allele frequency (MAF) less than 5%; (2) a call rate of less than 95%; or (3) a Hardy-Weinberg equilibrium exact test P-value \< 10^-6^. All participants were checked for duplicate genotype information across studies, and all participants with duplicated genotyping data were removed. Of participants with a high degree of relatedness estimated using identical by descent (IBD) indicated by 2^nd^-degree relation (proportion IBD \> 0.18) or closer in PLINK, all but the participant with the highest genotyping rate in each related cluster were removed, with ties broken randomly. In NACC/ADGC analyses, `r nacc_adgc_results[, .N]` variants and `r nacc_adgc_unrelated[, .N]` participants passed QC protocols. In the ROSMAP dataset, `r rosmap_results[, .N]` variants and `r rosmap_unrelated[, .N]` participants passed QC protocols. Thirteen ADNI participants were identified as NACC/ADGC duplicates and removed, leaving `r nacc_adgc_rosmap_adni_unrelated[adni == 1, .N]` participants for analysis. A total of **XXX (\~326)** ACT participants passed QC measures.

## **Identifying ethnic outliers**

In all cohorts, we performed principal component analysis (PCA) in PLINK v1.9 using a pruned subset of independent (linkage disequilibrium (LD) r^2^ \< 0.05) variants from each dataset merged to data from the 1000 Genomes Project Phase 3 (1000 Genomes, n = 2504) [@Abecasis2012]. All variants were checked to ensure that major and minor alleles matched in the study datasets and 1000 Genomes. We then plotted the standardized first and second principal components (PCs) for each participant using the ggplot2 R package in R version 4.0.3 [@rcoreteam2020]. All participants whose plot positions were located within a Euclidean distance of 0.3 from the mean plot positions of 1000 Genomes participants in the EUR superpopulation were considered to be of European descent and were included in analyses. We then re-ran PCA for the included participants and used included the first five PCs to use as covariates in regression models.

## **Statistical analyses**

### **Single-variant analyses**

We first performed single-variant association analyses in the NACC/ADGC participants using logistic regression in PLINK v1.9 and ordinal regression in R using the MASS package [@chang; @rcoreteam2020]. To create a dichotomous outcome variable for logistic regression, we counted participants with none or mild B-ASC as controls and those with moderate or severe B-ASC as cases. This decision was made based on previous studies that used these cut points and studies that found that moderate-to-severe B-ASC was associated with worse cognitive functioning [@ighodaro2017; @arvanitakis2016]. Covariates in the regression models included age at death, sex, ADGC cohort indicators, and the first five PCs of the genetic relatedness matrix. An additive mode of inheritance was assumed in all analyses. We used a Bonferroni-corrected threshold of P-value \< 5×10^-8^ for genome-wide significance and a predetermined "suggestive" threshold of P-value \< 1×10^-5^. Then, using PLINK v1.9, we clumped variants meeting this threshold to create a set of independent variants (defined as LD r^2^ \< 0.05). For suggestive variants in ordinal regression analyses, we tested the proportional odds assumption using Brant tests in the brant R package [@schlegel2020]. Finally, we examined single variants previously found to be putatively associated with B-ASC in Chou et al. (2013) to determine if any were validated in NACC/ADGC at the P-value \< 0.05 significance level [@chou2013].

In our ROSMAP analyses, we first performed single-variant regression analyses with variants identified as suggestively significant in the NACC/ADGC analyses, using a significance threshold of P-value \< 0.05 for these variants. We then performed ordinal and dichotomous genome-wide single-variant the procedures used in the NACC/ADGC analyses. Covariates included age at death, sex, cohort (ROS vs. MAP) and the first five PCs. We then performed a mega-analysis on both cohorts, including an indicator variable for ROSMAP participants. We also performed both fixed- and random-effects meta-analyses in PLINK v1.9 for individual ADGC and ROSMAP logistic GWAS results.

To seek replication of our results, we analyzed the top three variants from the NACC/ADGC and ROSMAP mega-analysis separately in the ADNI and ACT datasets. We then added the ADNI and ACT participants sequentially to our combined NACC/ADGC and ROSMAP dataset to perform additional mega-analyses.

### **Gene-based analyses**

# Placeholder

Following single-variant analyses, we performed gene-based and gene-set analyses using MAGMA [@leeuw2015]. First, we annoted variants to genes using Hg37 positions, and annotated all variants within a 1000-kilobase window of a gene as associated with it, producing a total of **XXXXX** genes with at least one annotated variant. We then performed competitive gene-based analyses in MAGMA. For the gene-set analysis, we used gene sets curated for Enrichment Map ([https://enrichmentmap.readthedocs.io](https://enrichmentmap.readthedocs.io/en/latest/index.html#)) that included gene sets based on multiple resources, including Reactome, Gene Ontology, the Pathway Interaction Database, NetPath, and the PANTHER database [@merico2010; ].

### **Colocalization analyses**

To investigate potential functional mechanisms of variants driving GWAS signals for B-ASC, we performed colocalization analyses for suggestive variants using the coloc R package and QTL summary statistic data from The Genotype-Tissue Expression (GTEx) project V8 publicly available data [@thegeno2013; @aguet2017; @giambartolomei2014; @gtexpor]. QTL are calculated in GTEx by performing single-variant analysis of gene expression for all variants within 1000 Kb of the transcription start or end site for each gene. First, we systematically checked to determine if suggestive variants from B-ASC analyses (P-value \< 1×10^-5^) are significant expression QTL (eQTL) or splicing QTL (sQTL) using GTEx summary statistics of participants of European descent. We then analyzed variants that were eQTL or sQTL in brain, nerve, blood, cultured cell lines, or vascular tissues (see Supplementary Table S2 for a full description of tissues used) and had nominal P-values less than ten times greater than the variant with the smallest nominal P-value for the associated QTL phenotype. We used default prior probabilities in the coloc package of P­~1~ = P~2~ = 1e-4 and P~12~ = 1e-5, meaning that each variant has a 1/10,000 prior probability of being associated with either trait, and a variant associated with either trait in turn has a 1/10 prior probability of being associated with both traits [@giambartolomei2014]. Because the coloc package can currently only analyze dichotomous or continuous variables, for QTL variants from ordinal B-ASC GWAS, we used variant P-values in the corresponding logistic analyses. A posterior probability of colocalization (PPH4) of 50% or greater was chosen to indicate evidence for colocalization.

### **Sensitivity analyses**

Given previously identified potential differences in clinical risk factors for B-ASC in participants stratified by age of death, we re-did our analyses in NACC/ADGC and ROSMAP using only participants with an age of death of eighty or above. To further assess the robustness of our results under different model assumptions, we performed several sensitivity analyses in our NACC/ADGC dataset using the binary B-ASC outcome variable. We first performed single-variant analyses on each ADGC cohort and then meta-analyzed in PLINK v1.9 rather than using fixed-effect cohort indicators in our regression models. For variants meeting our suggestive threshold, we included related participants and performed mixed-effects analyses with a random effect incorporating the kinship matrix estimated from KING in R using the GMMAT, GENESIS, and SNPRelate packages [@chen2020; @conomos2019; @gogarten; @zheng2012]. To overcome issues with computing PCs with samples with related participants, we used the PC-AiR method [@gogarten]. Finally we performed mediation analyses on the subset of our NACC/ADGC with clinical variables available to test whether diabetes mellitus or hypertension status mediate the association between suggestive genetic variants and B-ASC pathology [@ighodaro2017; @baron].

### **Testing Variants of Interest with WMH Volume**

We tested the top two variants from the single-variant meta-analysis for association with WMH volume in Alzheimer's Disease Neuroimaging Initiative (ADNI). ***Insert section here. Either Kwangsik or someone else at IU should write this section or Lincoln should discuss with Kwangsik on what methods were used for analysis.***

# Results

```{r prep-table-1}
# prepare table_1 data
source("code/paper_tables_figures/01_make_table_1.R")
# inline r code for paragraph
nacc_none <- table_1_data[Labels == "None", "NACC/ADGC"]
nacc_mild <- table_1_data[Labels == "Mild", "NACC/ADGC"]
nacc_moderate <- table_1_data[Labels == "Moderate", "NACC/ADGC"]
nacc_severe <- table_1_data[Labels == "Severe", "NACC/ADGC"]
rosmap_none <- table_1_data[Labels == "None", "ROSMAP"]
rosmap_mild <- table_1_data[Labels == "Mild", "ROSMAP"]
rosmap_moderate <- table_1_data[Labels == "Moderate", "ROSMAP"]
rosmap_severe <- table_1_data[Labels == "Severe", "ROSMAP"]
nacc_male <- nacc_adgc_unrelated[NPSEX == 1, round(.N * 100 / nacc_adgc_unrelated[, .N])]
rosmap_male <- rosmap_unrelated[msex == 1, round(.N * 100 / rosmap_unrelated[, .N])]
nacc_age <- nacc_adgc_unrelated[, round(mean(NACCDAGE), 1)]
rosmap_age <- rosmap_unrelated[, round(mean(age_death), 1)]

```

Of the `r nacc_adgc_unrelated[, .N]` NACC participants that met inclusion criteria for Sanalysis, `r nacc_none` had no B-ASC, `r nacc_mild` had mild B-ASC, `r nacc_moderate` had moderate B-ASC, and `r nacc_severe` had severe B-ASC (see **Table 1**). ROSMAP participants that met inclusion for analysis had comparatively less B-ASC pathology (P-value = 0.002): `r rosmap_none` had no B-ASC, `r rosmap_mild` had mild B-ASC, `r rosmap_moderate` had moderate B-ASC, and `r rosmap_severe` had severe B-ASC. NACC participants were also significantly more likely to be male (nacc_adgc_unrelated[NPSEX == 1, round(`r nacc_male`% vs.`r rosmap_male`%, P-value \< 0.001) and had younger ages at death on average (mean age of death `r nacc_age` vs. `r rosmap_age`, P-value \< 0.001) compared to ROSMAP participants. These comparative differences in the cohort demographics largely held in the secondary analysis that included only participants whose ages of death were eighty years or above, with the caveat that the difference in mean age of death was smaller between NACC and ROSMAP (87.7 vs. 90.7, P-value \< 0.001).

```{r include=TRUE}
# make table_1 object
table_1 <- autofit(table_1) %>% 
  padding(padding = 2, part = "all")
table_1
```

## Single-variant analyses

```{r single-variant-inline}
nacc_n_case <- nacc_adgc_unrelated[NACCARTE %in% 2:3, .N]
nacc_percent_case <- round(nacc_n_case / nacc_adgc_unrelated[, .N] * 100, 1)

rosmap_n_case <- rosmap_unrelated[arteriol_scler %in% 2:3, .N]
rosmap_percent_case <- round(rosmap_n_case / rosmap_unrelated[, .N] * 100, 1)

nacc_elovl4_p <- nacc_adgc_results[CHR == 6][grep("rs2603462", SNP), signif(P, 2)]
nacc_elovl4_or <- nacc_adgc_results[CHR == 6][grep("rs2603462", SNP), signif(OR, 3)]
nacc_sorcs3_p <- nacc_adgc_results[CHR == 10][grep("rs7902929", SNP), signif(P, 2)]
nacc_sorcs3_or <- nacc_adgc_results[CHR == 10][grep("rs7902929", SNP), signif(OR, 3)]

rosmap_sorcs3_p <- rosmap_results[CHR == 10][grep("rs7902929", SNP), signif(P, 2)]
rosmap_sorcs3_or <- rosmap_results[CHR == 10][grep("rs7902929", SNP), signif(OR, 3)]
```

```{r table-2-data}
source("code/paper_tables_figures/02_make_table_2_data.R")
```

In the logistic regression analysis, `r nacc_n_case` (`r nacc_percent_case`%) and `r rosmap_n_case` (`r rosmap_percent_case`%) of participants had either moderate or severe B-ASC and were counted as cases in NACC and ROSMAP, respectively. In the NACC/ADGC analysis, one locus, rs2603462 on chromosome 6q14.1, is genome-wide significantly associated with B-ASC (odds ratio [OR] = `r nacc_elovl4_or`, P-value = `r nacc_elovl4_p`). We identified 13 other loci that meet our suggestive association threshold of P-value \< 1×10^-5^, described in **Table 2**. No variants achieve genome-wide significance in any of the ROSMAP analyses, though **X** loci met our suggestive threshold in the primary logistic regression analysis. Of the 14 NACC/ADGC loci that met our suggestive threshold that were also tested in the ROSMAP analysis, rs7902929 on chromosome 10q25.1 was validated at P-value \< 0.05 level (NACC/ADGC OR = `r nacc_sorcs3_or`, P-value = `r nacc_sorcs3_p`; ROSMAP OR = `r rosmap_sorcs3_or`, P-value = `r rosmap_sorcs3_p`). For the other 13 loci, two ROSMAP ORs are \<1.00 and 11 have effect sizes in the same direction in effect as in NACC/ADGC, though of smaller magnitude. Results in the ordinal regressions were broadly similar to the logistic analyses in both cohorts. Of the top 14 loci in the NACC primary logistic regression, **X** also meet our suggestive threshold in the ordinal regression, along with **X** other loci. In the primary mega-analysis using logistic regression, **X** loci meet the suggestive threshold, with rs7902929 (OR = 1.58, P-value = 2.0 x 10-7) and rs2603462 (OR = 1.35, P-value = 3.0 x 10-7) producing the smallest P-values (see **Table 3**). Of the six variants found to be nominally associated with B-ASC by Chou et al. (2013), no variants were validated at the P-value \< 0.05 significance level in the NACC/ADGC cohort.

```{r table-2, include=TRUE}
# make table_2 object
table_2_data$`NACC P` <- format(table_2_data$`NACC P`, scipen=2)
table_2_data$`ROSMAP P` <- format(table_2_data$`ROSMAP P`, drop0trailing = TRUE)
table_2 <- flextable(table_2_data)

table_2 %>% 
  autofit() %>% 
  padding(padding = 2, part = "all")

```

```{r table-3, include = TRUE}

```

In the ADNI neuropathology cohort, both rs2603462 (OR = **X**, P = **X**) and rs7902929 (OR = **X**, P =**X** ) were valiedated at the *P* \< 0.05 significance level (see **Table 4**), while in the ACT neuropathology cohort, neither variant was validated.

In the secondary analysis using only participants with an age of death of 80 years or older, no variants achieved genome-wide significance (P-value \< 5x10^-8^). In the NACC/ADGC analyses, we identified 10 and 9 independent loci in the logistic and ordinal regression analyses, respectively, that met the suggestive threshold. Three loci were suggestively associated with B-ASC in both the logistic and ordinal analyses: rs4491854, an intergenic SNP near EOMES on chromosome 3 (logistic P-value = 8.0x10-7, ordinal P-value = 6.2x10-6); rs11928305, an intronic SNP in FAM19A1/TAFA1 on chromosome 3 (logistic P-value = 6.1x10-6, ordinal P-value = 4.0x10-7); and rs1956605, and intergenic SNP 500 Kb from NOVA1 on chromosome 14 (logistic P-value = 4.9x10-6, ordinal P-value = 6.3x10-6). No suggestive variants from the NACC/ADGC analyses were validated in the ROSMAP analyses at the P-value \< 0.05 significance threshold (see supplementary tables S3 and S4).

## Gene-based analyses

In the gene-based analyses, no genes achieved Bonferroni-adjusted significance (P-value \< 2.5x10-6).

## Colocalization analyses

In the mega-analyses, five variants met our criteria for colocalization analysis: rs2603462 colocalizes with B-ASC and ELOVL4 expression in the cerebellar hemisphere with PPH4 of 92.6%; rs1343705 colocalizes with expression of the non-coding gene RP11-408A13.3 in the cerebellum with a PPH4 of 87.2%; rs35010424 colocalizes with TAS2R5 expression in the hypothalamus with PPH4 of 84.4% and OR9N1P expression in the nucleus accumbens with PPH4 of 57.7%; rs34349961 colocalizes with SPRED2 expression in cultured fibroblasts with PPH4 of 74.6%; and rs6936285 colocalizes with CD83 in whole blood with PPH4 of 73.7% (see Table 4). In the NACC/ADGC analyses, rs2603462 and 3 suggestive loci are significant QTLs in GTEx and met our criteria for colocalization analysis across eleven phenotypes in nine tissues. Three of these loci colocalized with a PPH4 \> 50% across ten phenotypes. rs2603462 colocalizes with B-ASC and ELOVL4 expression in the cerebellar hemisphere with a PPH4 of 93.3%. rs2352974 is located in a gene-dense region of chromosome 3p21.31 and is a significant eQTL for DALRD3, FAM212A, MST1R, and TCTA in one tissue each, and a significant sQTL for RNF123 in four tissues; it colocalizes with B-ASC and seven of these phenotypes with PPH4 between 50.0% and 79.1%. No variants in the ROSMAP primary analyses met our criteria for colocalization analysis.

In the secondary NACC/ADGC analyses using only the oldest participants, eQTLs in GTEx were found for two genes, HLA-A and NMRAL2P. The HLA-A variant, rs9260090, is a significant eQTL in nine brain tissues, spanning the frontal cortex, basal ganglia, cingulate cortex, and cerebellum. Colocalization analysis in seven of these nine tissues showed evidence for colocalization with a posterior probability of colocalization (PPH4) greater than 50% and the brain cortex having the highest probability of colocalization (PPH4 = 82.1%). Evidence for colocalization is lowest in the hippocampus (PPH4 = 9.4%). The NMRAL2P variant, rs11718099, is an eQTL in the aortic artery, and colocalization analysis revealed a PPH4 of 86.2%. However, because rs11718099 was identified as being suggestively associated with B-ASC in the NACC/ADGC ordinal regression, its P-value from the logistic regression used for colocalization analysis is much larger (1.4 x 10^-3^ vs 8.7 x 10^-6^).

## Sensitivity analyses

Performing fixed-effects meta-analysis on the NACC/ADGC ADGC genotyping cohorts produced nearly identical effect sizes and P-values compared to including indicator variables for the non-reference cohorts in our primary analysis. Similarly, including the participants originally excluded due to relatedness did not produce notably different effect sizes or P-values for any of the suggestive variants. Finally, in the subset of our NACC/ADGC participants who had clinical data for diabetes and hypertension status (n = 1679), none of the suggestive variants were associated with diabetes or hypertension status at the P-value \< 0.05 level, and all variant effect sizes on B-ASC in regressions including diabetes or hypertension were within the 95% confidence intervals in the base model excluding them.

## WMH Volume Analyses

Insert section here.

# Conclusions

In this study, we performed the first GWAS of autopsy-proven B-ASC using neuropathology and genotype data from four neuropathology cohorts derived from multiple research centers. We found a significant association between one locus on Chromosome 6 and B-ASC in the NACC/ADGC cohort (*P* = `r nacc_elovl4_p`). This locus colocalizes with *ELOVL4* gene expression in GTEx (PPH4 = 93.3%), providing evidence that this locus may affect B-ASC risk through mediating *ELOVL4* expression (see **Figure X**). Another locus on Chromosome 10 near *SORCS3* suggestively associated with B-ASC was validated in the ROSMAP cohort. These variants' effects were successfully replicated in the ADNI cohort, but failed to replicate in the ACT cohort. We also found suggestive evidence for association between other loci and B-ASC risk in both cohorts and that some of these loci colocalize with gene eQTL and sQTL in GTEx.

Most genetic loci identified as being associated with B-ASC in NACC were not validated in ROSMAP. This could be potentially explained via the significantly different demographic attributes and study designs between the two cohorts, as ROSMAP participants were simultaneously older at death and had lower risk of B-ASC pathology, indicating that there may be unaccounted-for genetic confounding and differential selection bias between cohorts. For instance, ROS recruits from Catholic sisters and brothers, who are on average more highly educated than the general population. Each ADRC also has its own recruitment population. Differences in neuropathological grading of B-ASC in each cohort may also contribute to the heterogeneity of results, as B-ASC is graded globally in NACC but is graded in the basal ganglia in ROSMAP. Despite these limitations, one suggestive locus on chromosome ten identified in NACC/ADGC was validated in ROSMAP and had nearly the same affect size in each cohort (OR 1.58 vs. 1.60). Furthermore, 11 of 13 B-ASC-associated loci in NACC had affect sizes in the same direction in ROSMAP, which suggests that these loci may be associated with B-ASC but suffer from the "winner's curse" in the NACC/ADGC analyses.

*ELOVL4* codes for the elongation of very long chain fatty acids-4 protein, an elongase that synthesizes very long chain saturated and unsaturated fatty acids. Most research on *ELOVL4*has focused on its association with mendelian diseases affecting the visual and nervous systems, such as Stargardt-like macular dystrophy and spinocerebellar ataxia [@hopiavuori2019]. In GTEx, ELOVL4 is more highly expressed in the brain (median transcripts per kilobase million [TPM] 5.5-43.9) relative to most other tissues. We searched the GWAS Catalog and found two recent studies that found genome-wide significant associations between variants mapped to *ELOVL4* and multiple body weight-related phenotypes, including waist circumference adjusted for body mass index (BMI) (rs76567515, P = 2 x 10-12), waist-to-hip ratio (rs76567515, P = 3 x 10-10), and waist-to-hip ratio adjusted for BMI (rs1849275, P = 1 x 10-9) [@buniello2019; @zhu2020; @kichaev2019]. These variants are significant eQTLs for ELOVL4 in GTEx, but are each located \>200 Kb from the locus identified in our study and are not in LD with the lead variant (r2 \< 0.05). Additionally, another suggestive variant in NACC/ADGC, rs387083, is located between the free fatty acid receptor genes *FFAR1* and *FFAR3* on chromosome nineteen. These results suggest that fatty acid metabolism and signaling may play a role in B-ASC risk, and that ELOVL4 may potentially be related to B-ASC risk through its effects on BMI-related phenotypes.

The variant suggestively associated with B-ASC in the NACC/ADGC primary analyses that was subsequently validated in ROSMAP, rs7902929, is located approximately 212 Kb from the 3' end of the gene SORCS3. While the genes on which regulator intergenic variants exert their effects are not necessarily the closest genes, SORCS3 is the only protein-coding gene within a 1 megabase window from rs7902929, increasing the likelihood that the locus is functionally tied to it. SORCS3 codes for the sortilin-related VPS10 domain-containing receptor 3, a vacuolar protein expressed in the brain [@reitz2013; @wang2020]. Previous studies using candidate gene designs have provided tentative evidence that genetic variation in SORCS3 may be associated with AD [@reitz2013; @wang2020]. In the GWAS Catalog, SORCS3-mapped variants are significantly associated with multiple phenotypes, including depressive symptoms (rs1021363, P = 1 x 10-13), self-reported educational attainment (rs11599236, P = 1 x 10-13), and systolic blood pressure (rs191784289, P = 3 x 10-13) [@kichaev2019; @baselmans2019; @lee2018].

Variants identified as suggestively associated with B-ASC among participants aged 80 years or older at death were largely different than those identified in our primary analyses. One intronic locus of FAM19A1 was identified as suggestively associated with B-ASC in the NACC/ADGC dataset. This gene is highly and preferentially expressed in the brain [@thegeno2013]. FAM19A1 belongs to the family with sequence similarity 19 member A gene family, a five-member gene family with largely unknown function, though the family shares structural similarity to the CC-chemokine family [@wangyingbao2018]. A locus near the 3' end of another member of this family, FAM19A5, was identified as suggestively associated with B-ASC in the NACC/ADGC primary ordinal analysis. A recent study found that FAM19A5 is an adipokine that is capable of inhibiting post-injury neointima formation in injured blood vessels, and that FAM19A5 expression is downregulated in obesity.49 Another study found that FAM19A3, another gene homologue of FAM19A1, is highly expressed in microglia and helps attenuate cerebral ischemia [@shao2015].

Two loci suggestively associated with B-ASC in the analyses on older participants colocalized with gene expression with HLA-A and NRMAL2P in GTEx. HLA-A is a Major Histone Compatibility Complex (MHC) gene and is an integral component of the immune system. HLA-A has been found to be significantly associated with 120 complex traits in the GWAS Atlas [@watanabe2019]. While not identified in GWAS, certain HLA-A alleles have evidence of association with Alzheimer's disease (AD) risk [@cifuentes2014]. In one recent study, authors used ADNI data to investigate the association between AD HLA-A found that HLA-A variants were associated with atrophy of the left parahippocampus, right hippocampus, and right amygdala [@wang2017]. NRMAL2P is a long non-coding RNA (LncRNA) pseudogene that has previously been linked to gastric, colon, and esophageal cancers [@feng2020; @johnson2017; @mizumoto2019; ].

None of the effects of top loci in the NACC/ADGC GWAS were mediated by diabetes or hypertension status in NACC. This finding fails to help clarify the somewhat inconsistent evidence of association between these clinical risk factors and B-ASC pathology. However, this analysis was limited by a substantially smaller sample size (n = 1679) than our other analysis and the diagnostic criteria used, as the diagnostic variables in NACC used consisted of a mixture of patient self-reports, physician reporting, and use of hypertension or diabetes medications. Future studies employing causal inference methods such as two-sample Mendelian randomization may be able to provide clearer evidence for or against the genetic mediation of B-ASC risk by clinical risk factors.

In closing, we used autopsy-derived endophenotypes linked to genetic data in multiple cohorts to identify genetic loci associated with B-ASC pathology. We found one locus significantly associated with B-ASC that colocalizes with ELOVL4 gene expression in GTEx. This study provides first GWAS of autopsy-verified brain arteriolosclerosis pathology and contributes a growing body of literature that recognizes the importance of using autopsy-based cohorts to augment clinical diagnoses to study the genetics of dementia and cognitive impairment.

# Code Stash

```{r adni-mega}
# adni_mega <- fread("output/adni_npc/nacc_adgc_rosmap_adni_unrelated.assoc.logistic")
# adni_mega_top <- fread("output/adni_npc/nacc_adgc_rosmap_adni_unrelated.clumped")
# adni_mega_top <- adni_mega[SNP %in% adni_mega_top$SNP]
# setorder(adni_mega_top, P)
# adni_mega_top[, .(CHR, BP, SNP, P)] %>% kable() %>% kable_styling()
```

```{r act-meta}
# act_meta <- fread("output/act/act__nacc_adgc_rosmap_adni_unrelated.meta")
# setorder(act_meta, P)
# act_meta[P < 1e-5]
```

```{r magma}
# adni_mega_magma <- fread("output/magma/nacc_adgc_rosmap_adni_unrelated.genes.out")
# genes <- fread("raw_data/magma/NCBI37.3.gene.loc", header = F) %>% 
#   .[, .(V1, V6)] %>% 
#   setnames(c("V1", "V6"), c("GENE", "gene"))
# adni_mega_magma <- merge(adni_mega_magma, genes, "GENE")
# setorder(adni_mega_magma, P)
# head(adni_mega_magma, 20)
# 
# magma_gsa <- fread("output/magma/nacc_adgc_rosmap_adni_unrelated.gsa.out",
#                    skip = 4) %>%
#   setorder(P) %>%
#   .[, FULL_NAME := stri_replace_first_regex(FULL_NAME, "%.*", "")] %>%
#   .[, .(FULL_NAME, NGENES, P)]
# 
# head(magma_gsa, 10)
  
```

```{r locuszoom1, out.height=550, out.width=600}
# example of how to embed pdfs in rmd

# knitr::include_graphics("example.pdf")
```

# References
