---
output: word_document
bibliography: doc/references.bib
csl: doc/sage_vancouver_brackets.csl
link-citations: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE, include = FALSE)
```

```{r load-opts-packages-functions}
options(digits = 15)
library(pacman)
p_load(data.table, magrittr, stringi, flextable)
source("code/functions/strip_alleles.R")
nice_p <- function(x) {
  knitr:::format_sci(signif(x, 2), 'md')
}
nice_p <- Vectorize(nice_p)
```

```{r load-individual-level-data}
nacc <- readRDS("data/nacc/nacc.Rds")
load("data/nacc/nacc_80.RData")
nacc_80 <- nacc_adgc_unrelated
rosmap <- readRDS("data/rosmap/rosmap.Rds")
nacc_rosmap <- readRDS("data/nacc_rosmap/nacc_rosmap.Rds")
nacc_rosmap_adni <- readRDS("data/adni/nacc_rosmap_adni.Rds")
adni <- fread("raw_data/ADNI_NPC/ADNI_NPC_wCovariate.csv")
act <- readRDS("data/act/act.RData")
act_mega <- readRDS("data/act/act_mega.Rds")
act <- act[IID %in% act_mega[act == 1, IID]]
```

```{r load-single-variant-analysis-results, results = FALSE}
nacc_results <- fread("output/nacc/nacc.assoc.logistic")
nacc_80_results <- fread("output/nacc/nacc_80.assoc.logistic")
nacc_results[, SNP := strip_alleles(SNP)]
rosmap_results <- fread("output/rosmap/rosmap.assoc.logistic")
adni_replication_top <- fread("output/adni/top_replication.assoc.logistic")
act_results <- fread("output/act/act.assoc.logistic")

nacc_rosmap_results <- fread("output/nacc_rosmap/nacc_rosmap.assoc.logistic")
act_mega_results <- fread("output/act/act_mega.assoc.logistic")
```

```{r load-coloc-results}
act_mega_coloc <- fread("output/act/coloc_results.txt")
setnames(act_mega_coloc, 
         colnames(act_mega_coloc), 
         c("Gene", "Tissue", "CHR", "SNP", "SNPs_n", "PPH0", 
           "PPH1", "PPH2", "PPH3", "PPH4"))
act_mega_coloc[, PPH4 := round(PPH4 * 100, 1)]
```

# Title

"Genome-Wide Association Study of Brain Arteriolosclerosis"

# Authors

Lincoln M.P. Shade^1^, Yuriko Katsumata^1,2^, Timothy J. Hohman^3^, Julie A. Schneider^4^, Kwangsik Nho^5^, Andrew J. Saykin^5^, Shubhabrata Mukherjee^6^, Kevin L. Boehme^7^, John S.K. Kauwe^8^, Peter T. Nelson^2,9^, David W. Fardo^1,2^, Alzheimer's Disease Genetics Consortium, for the Alzheimer's Disease Neuroimaging Initiative^\*^

## Author Affiliations

^1^Department of Biostatistics, College of Public Health, University of Kentucky, Lexington, KY

^2^Sanders-Brown Center on Aging and Alzheimer's Disease Research Center, University of Kentucky, Lexington, KY

^3^Vanderbilt Memory & Alzheimer's Center, Department of Neurology, Vanderbilt University Medical Center, Nashville, Tennessee

^4^Departments of Neurology and Pathology, Rush University Medical Center, Chicago, IL

^5^Department of Radiology & Imaging Sciences, Indiana University School of Medicine, Indianapolis, IN

^6^Department of Medicine, University of Washington, Seattle, WA

^7^ARUP Laboratories, Salt Lake City, UT

^8^Office of the President, Brigham Young University--Hawaii, Laie, HI

^9^Pathology and Laboratory Medicine, University of Kentucky, Lexington, KY

^\*^Data used in preparation of this article were obtained from the Alzheimer's Disease Neuroimaging Initiative (ADNI) database (adni.loni.usc.edu). As such, the investigators within the ADNI contributed to the design and implementation of ADNI and/or provided data but did not participate in analysis or writing of this report. A complete listing of ADNI investigators can be found at: <http://adni.loni.usc.edu/wp-content/uploads/how_to_apply/ADNI_Acknowledgement_List.pdf>

## Corresponding Author

David W. Fardo

725 Rose Street, Room 201

Lexington, KY 40536-0082

[david.fardo\@uky.edu](mailto:david.fardo@uky.edu)

859-218-2070

\newpage

# Abstract

Brain arteriolosclerosis (B-ASC) is a small-vessel cerebrovascular disease characterized by pathologically altered arterioles. B-ASC is associated with cognitive impairment and increased likelihood of clinical dementia. To date no in-depth study has been conducted on genetic risk of autopsy-proven B-ASC. We performed a genome-wide association study (GWAS) of the B-ASC phenotype using multiple aged neuropathologic cohorts. Included in the study were participants with autopsy B-ASC endophenotype and genotype data available from NACC, ROSMAP, ADNI, and ACT data sets. Initial Stage 1 GWAS (*n* = `r nacc[, .N]`) and Stage 2 mega-analysis (*n* = `r nacc_rosmap[, .N]`) were performed using data from the two largest cohorts (NACC and ROSMAP). Replication of top variants and additional Stage 3 mega-analysis were performed incorporating two smaller cohorts (ADNI and ACT). Lead variants in the top two loci in the Stage 2 mega-analysis (rs7902929, *p* = `r nacc_rosmap_results[SNP == "rs7902929", nice_p(P)]` ; rs2603462, *p =* `r nacc_rosmap_results[SNP == "rs2603462", nice_p(P)]`) replicated in the first replication cohort (rs7902929, *p* = `r adni_replication_top[SNP == "rs7902929", signif(P, 2)]`; rs2603462, *p =* `r adni_replication_top[SNP == "rs2603462", signif(P, 2)]`). The rs2603462 lead variant colocalizes with *ELOVL4* expression in the cerebellum (posterior probability = 92.9%). Suggestive associations were also found for *SORCS1* and *SORCS3* regions. We thus identified putative loci associated with B-ASC risk but additional replication is required

# Keywords

VCID, SVD, arteriosclerosis, neuropathology, dementia, aging

\newpage

# Introduction

Brain arteriolosclerosis (B-ASC) is a subtype of cerebral small vessel neuropathologic change characterized by thickening of arteriole walls in the brain. These changes can include hypertrophy or atrophy of vascular smooth muscle and luminal extracellular deposition of collagen [@ighodaro2017; @grinberg2010; @buchman2013; @blevins2021]. B-ASC is commonly found in autopsied elderly individuals, with more than 50% of participants showing some B-ASC pathology in large autopsy studies [@ighodaro2017; @buchman2013; @chou2013 ]. B-ASC is often comorbid with multiple neuropathologies including limbic-predominant age-related TDP-43 encephalopathy (LATE), Alzheimer's disease (AD), hippocampal sclerosis (HS), micro-infarcts, and large-vessel infarcts [@ighodaro2017; @arvanitakis2016; @neltner2014 ]. B-ASC is also associated with cognitive decline after adjusting for age and the presence of other neuropathologies [@ighodaro2017; @arvanitakis2016]. Despite the clinical importance of B-ASC, its risk factors, other than age and sex @ighodaro2017, remain largely uncharacterized. Hypertension (HTN) and diabetes mellitus (DM) are established clinical risk factors for arteriolosclerosis in the kidneys, but their association with B-ASC is inconsistent; in an autopsy based cohort stratified by age at death (\<80 years vs. â‰¥80 years), HTN was significantly associated with B-ASC in the younger group only, and DM was not associated with B-ASC in either group [@ighodaro2017; @wu2005; @ikee2010; @cameron2006].

Genome-wide association studies (GWAS) have provided a powerful method for investigating genomic risk of complex diseases through analysis of millions of genetic variants with disease phenotypes of interest, and have to date identified many thousands of genetic variants associated with disease [@buniello2019]. For example, GWAS have idenfied genetic risk loci for stroke and for imaging-based phenotypes of cerebrovascular disease such as white matter hyperintensities (WMH) and brain infarcts [@chou2013; @beecham2014]. Studies of monogenic cerebrovascular disorders have identified genes associated with collagen and basement membrane formation, vascular smooth muscle, inflammation, and potassium ion channels [@blevins2021]. GWAS of HS have identified multiple ion channel-associated genes [@beecham2014] that have since been validated in gene-based studies [@katsumata2017].

There has been no published GWAS for autopsy-proven B-ASC to date [@traylor2016; @malik2018; @chauhan2019]. Given the unique and complex structures of brain arterioles and their associated structures, including astrocytes and other components of the blood-brain barrier, there is reason to suspect that the genomic risk for B-ASC may not be wholly shared with other cerebrovascular phenotypes @blevins2021. In one study of Religious Orders Study/Memory and Aging Project (ROSMAP) participants, 167 independent genetic variants previously meeting genome-wide significance threshold (*p* \< $5\times10^{-8}$) in GWAS of stroke or stroke risk factors were tested for association with B-ASC pathology [@chou2013]. The authors found six variants nominally associated with B-ASC at the 0.01 \< *p* \< 0.05 significance level [@chou2013]. Another study found that an HS risk variant located in *ABCC9* was nominally associated with B-ASC [@ighodaro2017]. Investigating genomic risk factors of B-ASC at the genome-wide level may provide important insights into its pathophysiological development as well as its relationship to neuroimaging and other neuropathological phenotypes.

In the present study, we conducted a GWAS using B-ASC pathology as the endophenotype in four autopsy cohorts with genotype data available.


# Methods

A study design outline is shown in **Figure 1**. In Stage 1, a GWAS was performed on data from the National Alzheimer's Coordinating Center (NACC) Neuropathology data set linked to genotype data from the Alzheimer's Disease Genetics Consortium (ADGC). In Stage 2, top variants from the NACC GWAS were first analyzed in ROSMAP, and then the two data sets were merged and mega-analyzed. In Stage 3, we attempted to replicate the top variants from Stage 2 in two smaller autopsy cohorts consisting of Alzheimer's Disease Neuroimaging Inititative (ADNI) and Adult Changes in Thought (ACT) participants. To investigate potential biologically functional correlates to disease risk, we then perform colocalization analysis on B-ASC associated variants identified as quantitative trait loci (QTL) using data from the Genetic Tissue Expression (GTEx) Project and gene-based association analyses [@thegeno2013]. To investigate the possibility of HTN and DM acting as mediators for B-ASC risk variants, we also performed mediation analyses on a subset of NACC participants with clinical data available.

[insert **Figure 1**]

## Study Participants

```{r}
nacc_ids_pass_ex <- fread("data/nacc/nacc_ids_pass_exclusion.txt") %>% 
  .[, .N]
rosmap_np <- fread("data/rosmap/rosmap_ids_np.txt") %>% 
  .[, .N]
```

For the Stage 1 GWAS, neuropathology data from participants in \>30 National Institute on Aging-funded Alzheimer's Disease Research Centers (ADRCs) studies in the NACC Neuropathology Data Set were linked to ADGC genotype data [@welcome; @besser2018]. Each ADRC has its own recruitment strategies, populations, and study design, and data are collected and aggregated by NACC. Participants diagnosed with any of 19 unusual neurological conditions were then excluded because B-ASC may represent something different in the presence of these rare diseases  (see **Supplementary Table S1** for full exclusion criteria). Similar exclusion criteria were not applied to other studies' participants due to lack of data availability. A total of `r nacc_ids_pass_ex` participants had both B-ASC neuropathology and genotype data available and passed initial inclusion criteria.

The ROSMAP study has been previously described in detail and consists of harmonized data from two longitudinal cohorts: The Religious Orders Study (ROS) and the Memory and Aging Project (MAP) [@bennett2018 ]. ROS began in 1994 and recruited older Catholic Priests and Sisters from around the United States. MAP began in 1997 and recruited older adults who at the time had no diagnosis of dementia. A total of `r rosmap_np` ROSMAP participants had both autopsy and genotype data available.

The ADNI ([adni.loni.usc.edu](http://adni.loni.usc.edu/)) was launched in 2003 as a public-private research partnership, led by Principal Investigator Michael W. Weiner, MD. The primary goal of ADNI has been to test whether serial magnetic resonance imaging (MRI), positron emission tomography (PET), other biological markers, and clinical and neuropsychological assessment can be combined to measure the progression of mild cognitive impairment (MCI) and early AD. A subset of ADNI participants undergo autopsy and receive neuropathological phenotyping. A total of 60 ADNI participants had both B-ASC and genotype data available.

The ACT study began in 1994 and recruited residents in the greater Seattle area aged 65 years and older without dementia at time of enrollment [@kukull2002]. The goal of the study has expanded to include three cohorts and continuous enrollment using the same enrollment criteria and has a current total of 4,960 particpants across all three cohorts. A total of `r act[, .N] + 6` ACT participants had both B-ASC and genotype data available.

## Definitions of B-ASC variables used

In the NACC Neuropathology Data Set, B-ASC was graded as an ordinal variable with possible values of 0 (none), 1 (mild), 2 (moderate), or 3 (severe). Grading was performed by trained neuropathologists at each ADRC and was a global rating, meaning no specific instruction was given to examine specific brain regions for B-ASC. The B-ASC variable in ROSMAP was graded on B-ASC histological changes exclusively in the basal ganglia. Vessel intimal pathology is first graded on a scale from 0 (none) to 6 (severe), and then collapsed to a four-level ordinal variable with the same labels as in NACC [@buchman2011]. ADNI and ACT autopsy procedures followed the NACC Neuropathology codebook [@franklin2015], though B-ASC in ACT is labeled 1-4 rather than 0-3.

## Identifying ethnic outliers

In all cohorts, principal component analysis (PCA) in PLINK v1.9 [@chang2015] was performed with a pruned subset of independent (linkage disequilibrium (LD) r^2^ \< 0.05) variants from each data set merged to data from the 1000 Genomes Project Phase 3 (1000 Genomes, n = 2504) [@Abecasis2012]. All variants were checked to ensure that major and minor alleles matched in the study data sets and 1000 Genomes. The standardized first and second principal components (PCs) were then plotted for each participant using the ggplot2 R package in R version 4.0.4 [@rcoreteam2020]. All participants whose plot positions were located within a Euclidean distance of 0.35 from the mean plot position of 1000 Genomes participants in the EUR superpopulation were considered to be of European descent and were included in analyses. We then re-ran PCA for the included participants and included the first five PCs as covariates in regression models.

## Quality control (QC) of participants and genotype data

Imputation for ADGC was performed using the Haplotype Reference Consortium (HRC) panel [@mccarthy2016] (workflow can be found at <https://github.com/KBoehme/ADGC_HRC_MERGE>). The main ROSMAP genotype QC workflow has been previously described @dumitrescu2020. Briefly, imputation was performed for ROSMAP on the Michigan Imputation Server @das2016 using the HRC panel. Imputation for ACT was performed on the Michigan Imputation Server using the Trans-Omics for Precision Medicine (TOPMed) reference panel [@fuchsberger2015; @taliun2019]. We performed additional standard QC procedures on all genotyping data using PLINK v1.9 and KING [@chang2015; @chang; @purcell2007; @manichaikul2010; @marees2018]. Variants were excluded if they had (1) a minor allele frequency (MAF) less than 5%; (2) a call rate of less than 95%; or (3) a Hardy-Weinberg equilibrium exact test *p* \< $1\times10^{-6}$ using all participants in a cohort.

Participants were removed if they had a genotype missingness rate above 5% or a heterozygosity rate more than 3 standard deviations above or below the mean of their cohort. All participants were checked for duplicate genotype information across studies. All participants with duplicate information within one cohort were removed, while participants with duplicate information across cohorts were preferentially kept in the data set analyzed first and removed from cohorts used later in the study (*i.e.* NACC \> ROSMAP \> ADNI and ACT). Of participants with a high degree of relatedness estimated using identical by descent (IBD) indicated by 2^nd^-degree relation (proportion IBD \> 0.18) or closer in PLINK, all but the participant with the highest genotyping rate in each related cluster were removed, with ties broken randomly. In NACC analyses, `r nacc_results[, .N]` variants and `r nacc[, .N]` participants passed QC protocols. In the ROSMAP data set, `r rosmap_results[, .N]` variants and `r rosmap[, .N]` participants passed QC protocols. Thirteen ADNI participants were identified as NACC duplicates and removed, leaving `r nacc_rosmap_adni[adni == 1, .N]` participants for analysis. A total of `r act[, .N]` ACT participants passed QC measures.

## Statistical analyses

### Single-variant analyses

Single-variant association analyses for NACC were performed using logistic regression in PLINK v1.9 and ordinal regression in R using the MASS package [@chang; @rcoreteam2020]. To create a dichotomous outcome variable for logistic regression, participants with none or mild B-ASC were counted as controls and those with moderate or severe B-ASC as cases. This decision was made based on previous studies that used these cut points and found that moderate-to-severe B-ASC was associated with worse cognitive functioning [@ighodaro2017; @arvanitakis2016]. Covariates in the regression models included age at death, sex, ADGC cohort indicators, and the first five PCs of the genetic relatedness matrix. An additive mode of inheritance was assumed in all analyses. A Bonferroni-corrected threshold of *p* \< $5\times10^{-8}$ was used for genome-wide significance and a predetermined threshold of *p* \< $1\times10^{-5}$ for "suggestive" association meriting further investigation. Then, using PLINK v1.9, variants meeting the suggestive threshold were clumped to create a set of independent variants (defined as LD *r^2^* \< 0.05). For suggestive variants in ordinal regression analyses, the proportional odds assumption was tested using the Brant test in the brant R package [@schlegel2020]. Finally, we examined single variants previously found to be putatively associated with B-ASC in Chou et al. (2013) [@chou2013] to determine if any were validated in NACC at the *p* \< 0.05 significance level.

In the ROSMAP analyses, single-variant regression analyses were performed using variants identified as suggestively significant in the NACC analyses, using a significance threshold of *p* \< 0.05 for these variants. Covariates included age at death, sex, study (ROS vs. MAP) and the first five PCs. We then performed a mega-analysis on both cohorts, including an indicator variable for ROSMAP participants. We also performed both fixed- and random-effects meta-analyses in PLINK v1.9 for ADGC and ROSMAP logistic GWAS results.

To seek replication of our results, variants with NACC + ROSMAP mega-analytic *p* \< $1\times10^{-6}$ were analyzed separately in the ADNI and ACT data sets. The ADNI and ACT participants were then added to the combined NACC and ROSMAP data set to perform a combined mega-analyss. A total of `r act_mega_results[, .N]` variants were shared between the NACC, ROSMAP, ADNI, and ACT data sets that passed QC measures.

### Gene-based analyses

```{r}
magma <- fread("output/magma/act_mega.genes.out")
```

Following single-variant analyses, we performed gene-based analyses using MAGMA [@leeuw2015]. Variants were mapped to genes within 1000 kilo-basepairs (kb) using the Genome Reference Consortium Human Build 37 (GRCh37/hg19) positions available on the MAGMA website (<https://ctg.cncr.nl/software/magma>), producing a total of `r magma[, .N]` genes with at least one annotated variant. We then performed gene-based analyses in MAGMA using a Bonferroni-corrected significance threshold of *p* \< $2.5\times 10^{-6}$.

### Colocalization analyses

To investigate potential functional mechanisms of variants driving GWAS signals for B-ASC, we performed colocalization analyses for suggestive variants using the coloc R package and QTL summary statistic data from The GTEx project V8 publicly available data (accessible at <https://www.gtexportal.org/home/datasets>) [@thegeno2013; @aguet2017; @giambartolomei2014; @gtexpor]. QTL are calculated in GTEx by performing single-variant analysis of gene expression for all variants within 1000 kb of the transcription start or end site for each gene. First, we systematically checked to determine if suggestive variants from our Stage 3 mega-analysis (*p* \< $1\times10^{-5}$) were significant expression QTL (eQTL) or splicing QTL (sQTL) across 52 tissues using GTEx summary statistics of participants of European descent. We then performed colocalization analysis on each significant phenotype/tissue combination using default prior probabilities in the coloc package at $p_1 = p_2 = 1\times10^{-4}$ and $p_{12} = 1\times10^{-5}$. These parameter choices indicate that each variant used has a 1/10,000 prior probability of being a B-ASC risk variant or a QTL, and a variant that is either a B-ASC risk variant or a QTL has a 1/10 prior probability of being both [@giambartolomei2014]. Because the coloc package can currently only analyze dichotomous or continuous variables, we did not analyze variants from the ordinal regression analyses. A posterior probability of colocalization (PPH4) of 50% or greater was chosen to indicate evidence for colocalization.

### Mediation analyses

To test the hypotheses that variants associated with B-ASC risk may be mediated by HTN or DM, we performed mediation analyses using R on the subset of NACC participants with clinical variables available [@ighodaro2017; @baron]. Participants were labeled as DM or HTN cases if they had at least one of 1) self-reported diagnosis, 2) clinician-reported diagnoses, or 3) reported use of DM or HTN medication on their most recent clinical visit prior to death.

```{=html}
<!--
### **Testing Variants of Interest with WMH Volume**

We tested the top two variants from the single-variant meta-analysis for association with WMH volume in Alzheimer's Disease Neuroimaging Initiative (ADNI). ***Insert section here. Either Kwangsik or someone else at IU should write this section or Lincoln should discuss with Kwangsik on what methods were used for analysis.***
-->
```
### Sensitivity analyses

Given previously identified potential differences in clinical risk factors for B-ASC in participants stratified by age at death, we re-analyzed our analyses in NACC and ROSMAP using only participants with an age at death of 80 or above. To further assess the robustness of our results under different model assumptions, we performed several sensitivity analyses in our NACC data set using the binary B-ASC outcome variable. We first performed single-variant analyses on each ADGC cohort and then meta-analyzed in PLINK v1.9 rather than using fixed-effect cohort indicators in our regression models. For variants meeting our suggestive threshold, we included related participants and performed mixed-effects analyses with a random effect incorporating the kinship matrix estimated from KING in R using the GMMAT, GENESIS, and SNPRelate packages [@chen2020; @conomos2019; @gogarten; @zheng2012]. To overcome issues with computing PCs with samples with related participants, we used the PC-AiR method [@gogarten].

# Results

```{r prep-table-1}
# prepare table_1 data
source("code/manuscript/01_table_1.R")
# inline r code for paragraph
nacc_none <- table_1_data[Labels == "None", "NACC"]
nacc_mild <- table_1_data[Labels == "Mild", "NACC"]
nacc_moderate <- table_1_data[Labels == "Moderate", "NACC"]
nacc_severe <- table_1_data[Labels == "Severe", "NACC"]
rosmap_none <- table_1_data[Labels == "None", "ROSMAP"]
rosmap_mild <- table_1_data[Labels == "Mild", "ROSMAP"]
rosmap_moderate <- table_1_data[Labels == "Moderate", "ROSMAP"]
rosmap_severe <- table_1_data[Labels == "Severe", "ROSMAP"]
nacc_male <- nacc[NPSEX == 1, round(.N * 100 / nacc[, .N])]
rosmap_male <- rosmap[msex == 1, round(.N * 100 / rosmap[, .N])]
nacc_age <- nacc[, round(mean(NACCDAGE), 1)]
rosmap_age <- rosmap[, round(mean(age_death), 1)]
act_none <- table_1_data[Labels == "None", "ACT"]

```

Of the `r nacc[, .N]` NACC participants that met inclusion criteria for analysis, `r nacc_none` had no B-ASC, `r nacc_mild` had mild B-ASC, `r nacc_moderate` had moderate B-ASC, and `r nacc_severe` had severe B-ASC (see **Table 1**). ROSMAP participants that met inclusion for analysis had comparatively less B-ASC pathology (*p* = 0.002): `r rosmap_none` had no B-ASC, `r rosmap_mild` had mild B-ASC, `r rosmap_moderate` had moderate B-ASC, and `r rosmap_severe` had severe B-ASC. NACC participants were also significantly more likely to be male (`r nacc_male`% vs.`r rosmap_male`%, *p* \< 0.001) and had younger ages at death on average (mean age at death `r nacc_age` vs. `r rosmap_age`, *p* \< 0.001) compared to ROSMAP participants. ADNI had the highest proportion of male participants of any cohort at `r round(adni[NPSEX == 1, .N] / adni[, .N] * 100, 1)` %, while ACT had the lowest percentage at participant with no B-ASC with `r act_none` participants.

## Single-variant analyses

```{r single-variant-inline}
nacc_n_case <- nacc[NACCARTE %in% 2:3, .N]
nacc_percent_case <- round(nacc_n_case / nacc[, .N] * 100, 1)

rosmap_n_case <- rosmap[arteriol_scler %in% 2:3, .N]
rosmap_percent_case <- round(rosmap_n_case / rosmap[, .N] * 100, 1)

nacc_elovl4_p <- nacc_results[CHR == 6][grep("rs2603462", SNP), nice_p(P)]
nacc_elovl4_or <- nacc_results[CHR == 6][grep("rs2603462", SNP), signif(OR, 3)]
nacc_sorcs3_p <- nacc_results[CHR == 10][grep("rs7902929", SNP), nice_p(P)]
nacc_sorcs3_or <- nacc_results[CHR == 10][grep("rs7902929", SNP), signif(OR, 3)]

rosmap_sorcs3_p <- rosmap_results[CHR == 10][grep("rs7902929", SNP), nice_p(P)]
rosmap_sorcs3_or <- rosmap_results[CHR == 10][grep("rs7902929", SNP), signif(OR, 3)]

rosmap_clump_n <- fread("output/rosmap/rosmap.clumped") %>% 
  .[, .N]

nacc_ordinal <- fread("output/nacc/ordinal_results.txt")
nacc_clump <- fread("output/nacc/nacc.clumped")
nacc_rosmap_clump <- fread("output/nacc_rosmap/nacc_rosmap.clumped")
```

```{r table-2-data}
source("code/manuscript/02_table_2_data.R")
# make table_2 object
table_2_data$`NACC P` <- format(table_2_data$`NACC P`, scipen=2)
table_2_data$`ROSMAP P` <- format(table_2_data$`ROSMAP P`, drop0trailing = TRUE)
table_2 <- flextable(table_2_data)

nacc_80_clump <- fread("output/nacc/nacc_80.clumped")
```

In the Stage 1 analyses, `r nacc_n_case` (`r nacc_percent_case`%) and `r rosmap_n_case` (`r rosmap_percent_case`%) of participants had either moderate or severe B-ASC and were counted as cases in NACC and ROSMAP, respectively. In the Stage 1 NACC GWAS (*n* = `r nacc[, .N]`), one locus on Chromosome 6q14.1 with sentinal variant rs2603462 was genome-wide significantly associated with B-ASC (odds ratio [OR] = `r nacc_elovl4_or`, *p* = `r nacc_elovl4_p`). We identified 13 other loci that met our suggestive association threshold of *p* \< $1\times10^{-5}$, described in **Table 2**. Of the 14 NACC loci that met our suggestive threshold, rs7902929 on Chromosome 10q25.1 was validated at *p* \< 0.05 level in ROSMAP (NACC OR = `r nacc_sorcs3_or`, *p* = `r nacc_sorcs3_p`; ROSMAP OR = `r rosmap_sorcs3_or`, *p* = `r rosmap_sorcs3_p`). For the other 13 loci, two variants did not pass QC in ROSMAP, two had non-significant ROSMAP ORs \< 1.00, and 9 had non-significant effect sizes in the same direction as in NACC. Results in the ordinal regressions were broadly similar to the logistic analyses in both cohorts. Of the top 14 loci in the NACC logistic regression analysis, four also met our suggestive threshold in the ordinal regression analysis, and ten had *p* \< $1\times10^{-4}$ (Supplementary **Table S2**) Of the six variants found to be nominally associated with B-ASC by Chou et al. (2013) [@chou2013], no variants were validated at the *p* \< 0.05 significance level in the NACC cohort.

```{r table-3-data}
source("code/manuscript/03_table_3_data.R")
adni_elovl4_or_95ci <- table_3_data[SNP == "rs2603462", adni_or_95ci]
adni_elovl4_p <- table_3_data[SNP == "rs2603462", P_ADNI]
adni_sorcs3_or_95ci <- table_3_data[SNP == "rs7902929", adni_or_95ci]
adni_sorcs3_p <- table_3_data[SNP == "rs7902929", P_ADNI]
mega_sorcs3_or <- nacc_rosmap_results[SNP == "rs7902929", round(OR, 2)]
mega_sorcs3_p <- nacc_rosmap_results[SNP == "rs7902929", nice_p(P)]
mega_elovl4_or <- nacc_rosmap_results[SNP == "rs2603462", round(OR, 2)]
mega_elovl4_p <- nacc_rosmap_results[SNP == "rs2603462", nice_p(P)]
```

In the Stage 2 NACC and ROSMAP mega-analysis (*n* = `r nacc_rosmap[, .N]`), `r nacc_rosmap_clump[, .N]` loci met the suggestive significance threshold, with rs7902929 (OR = `r mega_sorcs3_or`, *p* = `r mega_sorcs3_p`) and rs2603462 (OR = `r mega_elovl4_or`, *p* = `r mega_elovl4_p`) producing the smallest *p*-values. Findings from the Stage 2 mega-analysis are illustrated in **Figure 2**. Because the ADNI (*n* = `r adni[, .N]`) and ACT (*n* = `r act[, .N]`) sample sizes were much smaller than those NACC and ROSMAP, we limited the replication attempts in these cohorts to variants meeting a more stringent threshold of *p* \< $1\times10^{-6}$, which two variants, rs2603462 and rs7902929, met. In the ADNI neuropathology cohort, effects of both rs2603462 (OR [95% confidence interval (CI)] =`r adni_elovl4_or_95ci`, P =`r adni_elovl4_p`) and rs7902929 (OR [95% CI] =`r adni_sorcs3_or_95ci`, P =`r adni_sorcs3_p` ) were replicated at the *P* \< 0.05 significance level (**Table 3**). In the ACT neuropathology cohort, neither variant was replicated, and both variants had non-signifant effects in the opposite direction than in the other cohorts.

[insert **Figure 2**]

In the Stage 3 mega-analysis combining the NACC, ROSMAP, ADNI, and ACT cohorts, no variants met the genome-wide significance threshold. A total of nine loci met the threshold for suggestive significance (**Table 4**), of which five were identified in Stage 2, while four were novel loci. Both rs2603462 and rs7902929 met the suggestive threshold (**Figure 3**).

```{r table-4-data, include=TRUE}
source("code/manuscript/04_table_4_data.R")
```

## Gene-based analyses

```{r magma-results}
gene_labels <- fread("raw_data/magma/NCBI37.3.gene.loc", header = FALSE) %>% 
  .[, .(V1, V6)] %>% 
  setnames(colnames(.), c("GENE", "Gene"))

magma_genes <- merge(magma, gene_labels, "GENE") %>% 
  setorder(P)
magma_sorcs1 <- magma_genes[Gene == "SORCS1", nice_p(P)]
magma_sorcs3 <- magma_genes[Gene == "SORCS3", nice_p(P)]
```

In the gene-based analyses, no genes achieved Bonferroni-adjusted significance (*p* \< $2.5\times10^{-6}$). *SORCS1* (*p* = `r magma_sorcs1`) achieved the smallest *p*-value of any gene, while the adjacent *SORCS3* gene *p*-value was somewhat larger (*p* = `r magma_sorcs3`).

## Colocalization analyses

```{r}
coloc_ffar3 <- act_mega_coloc[SNP == "rs12459138" & Tissue == "Adipose_Visceral_Omentum", PPH4]
coloc_elovl4 <- act_mega_coloc[SNP == "rs2603462", PPH4]
coloc_lincrna_cereb <- act_mega_coloc[SNP == "rs157913" & Tissue == "Brain_Cerebellum", PPH4]
coloc_lincrna_hypot <- act_mega_coloc[SNP == "rs157913" & Tissue == "Brain_Hypothalamus", PPH4]
coloc_wasf3_caud <- act_mega_coloc[SNP == "rs61944465" & Tissue == "Brain_Caudate_basal_ganglia", PPH4]
coloc_wasf3_fibr <- act_mega_coloc[SNP == "rs61944465" & Tissue == "Cells_Cultured_fibroblasts", PPH4]
coloc_nsd1 <- act_mega_coloc[SNP == "rs4608937"][1, PPH4]
```

In the mega-analyses, five suggestive variants were significant QTLs across nine phenotypes and 12 tissues in GTEx, leading us to perform colocalization analyses for a total of 18 phenotype-tissue combinations. Of these, nine met our evidential threshold of PPH4 \> 50% for colocalization. The strongest evidence of colocalization was found in *FFAR3* expression in visceral adipose tissue with PPH4 = `r coloc_ffar3`%, followed by expression of the long intergenic non-coding RNA (lincRNA) AC058791.1 in the cerebellum and hypothalamus with PPH4 of `r coloc_lincrna_cereb`% and `r coloc_lincrna_hypot`%, respectively. *ELOVL4* expression in the cerebellar hemisphere colocalizes with B-ASC risk with a PPH4 of `r coloc_elovl4`%. *WASF3* expression colocalizes with B-ASC risk across three brain regions, including the basal ganglia with PPH4 = `r coloc_wasf3_caud`%, and preferential splicing of *WASF3* in fibroblasts colocalizes with `r coloc_wasf3_fibr`% (**Table 5**).

## Mediation analyses

```{r}
nacc_dm <- fread("data/nacc/nacc_dm.txt")
nacc_htn <- fread("data/nacc/nacc_htn.txt")
```

In the subset of our NACC participants who had clinical data for DM (*n* = `r nacc_dm[, .N]`) and HTN (*n* = `r nacc_htn[, .N]`) status, none of the suggestive variants were associated with HTN or DM at the *p* \< 0.05 level, and all variant effect sizes in regressions including HTN or DM were within the 95% confidence intervals in the base model excluding them.

## Sensitivity analyses

In the analysis using only participants with an age at death of 80 years or older, no variants achieved genome-wide significance. In the NACC analysis, we identified `r nrow(nacc_80_clump)` independent loci that met the suggestive threshold. No suggestive variants from the NACC analyses were validated in the ROSMAP analyses at the *p* \< 0.05 significance threshold (Supplementary **Table S2**). Performing fixed-effects meta-analysis on the NACC ADGC genotyping cohorts produced nearly identical effect sizes and *p*-values compared to including indicator variables for the non-reference cohorts in our primary analysis. Similarly, including the participants originally excluded due to relatedness did not produce notably different effect sizes or *p*-values for any of the suggestive variants.

# Discussion

The present study constituted the first GWAS of autopsy-proven B-ASC using neuropathology and genotype data from four large autopsy cohorts. A significant association was found between one locus on Chromosome 6 and B-ASC in the Stage 1 NACC GWAS (*p* = `r nacc_elovl4_p`). In the Stage 3 mega-analysis of all four cohorts, this locus colocalized with *ELOVL4* gene expression in GTEx (PPH4 = `r coloc_elovl4`%), providing evidence that this locus may affect B-ASC risk through mediating *ELOVL4* expression. Another locus on Chromosome 10 near *SORCS3* suggestively associated with B-ASC was validated in the ROSMAP cohort. These variants' effects were replicated in the ADNI cohort, but failed to replicate in the ACT cohort. We also found suggestive evidence for association between other loci and B-ASC risk in both cohorts and that some of these loci colocalize with gene eQTL and sQTL in the GTEx data set.

*ELOVL4* codes for the elongation of very long chain fatty acids-4 protein, an "elongase" enzyme that catalyzes the synthesis of very long chain fatty acids. Much of the research on *ELOVL4* has focused on its association with Mendelian diseases affecting the visual and nervous systems, such as Stargardt-like macular dystrophy and spinocerebellar ataxia [@hopiavuori2019]. In GTEx, *ELOVL4* is more highly expressed in the brain (median transcripts per kilobase million [TPM] 5.5-43.9) relative to most other tissues. Two recent studies found genome-wide significant associations between variants mapped to *ELOVL4* and multiple body weight-related phenotypes, including waist circumference adjusted for body mass index (BMI) (rs76567515, (*p* = $2\times10^{-12}$), waist-to-hip ratio (rs76567515, *p* = $3\times10^{-10}$), and waist-to-hip ratio adjusted for BMI (rs1849275, *p* = $1\times10^{-9}$) [@buniello2019; @zhu2020; @kichaev2019]. These variants were also significant eQTLs for *ELOVL4* in GTEx, but are each located \>200 kb from the locus identified in our study and are not in LD with the lead variant (*r^2^* \< 0.05). Interestingly, another gene, *BCKDHB*, is located closer to the sentinal variant than *ELOVL4*. *BCKDHB* is one of three genes responsible for the monogenic metabolism disease Maple Syrup Urine Disease [@strauss1993]. Additionally, another suggestive variant in NACC, rs387083, is located between the free fatty acid receptor genes *FFAR1* and *FFAR3* on Chromosome 19. This locus acts as an eQTL of *FFAR3* in the visceral omentum adipose. Collectively, these results suggest that fatty acid metabolism and signaling may play a role in B-ASC risk.

The variant suggestively associated with B-ASC in the NACC primary analyses subsequently validated in ROSMAP, rs7902929, is located approximately 212 kb from the 3' end of the gene *SORCS3*. While the genes on which intergenic variants exert their effects are not necessarily the closest genes, *SORCS3* is the only protein-coding gene within a 1 megabase window from rs7902929, increasing the likelihood that the locus is functionally tied to it. *SORCS3* codes for the sortilin-related VPS10 domain-containing receptor 3, a vacuolar protein expressed in the brain [@reitz2013; @wang2020]. Previous studies using candidate gene designs have provided tentative evidence that genetic variation in *SORCS3* may be associated with AD [@reitz2013; @wang2020]. In the GWAS Catalog, *SORCS3*-mapped variants are significantly associated with multiple phenotypes, including depressive symptoms (rs1021363, *p* = $1\times10^{-13}$), self-reported educational attainment (rs11599236, *p* = $1\times10^{-13}$), and systolic blood pressure (rs191784289, P = $3\times10^{-13}$) [@kichaev2019; @baselmans2019; @lee2018].

Additionally, a member of the VPS10 family, *SORCS1*, achieved the smallest *p*-value in the gene-based analysis. We performed a *post-hoc* analysis of this region and identified two independent loci with lead *p*-values just above the suggestive threshold driving this effect: one locus in LD with rs7902929 and one not (*p* = $1.2\times10^{-5}$). We examined the lead variant in the second locus and identified a consistent effect across all four cohorts that was nominally significant in NACC (*p* = $1\times10^{-4}$) and ACT (*p* = 0.001), though not in ROSMAP (*p* = 0.25) or ADNI (*p* = 0.07). Previous studies have identified *SORCS1* as a potential risk gene for AD [@reitz2013; @wang2020]. Therefore, it is possible that the VPS10 family influences B-ASC risk through AD risk, or vice versa, though further work should be done to clarify the possible role of the VPS10 family on both phenotypes.

Statistical modeling indicated that none of the effects of top loci in the NACC GWAS appeared to be mediated by HTN or DM status in NACC. This finding has some caveats, including the inconsistent evidence of association between these clinical risk factors and B-ASC pathology [@ighodaro2017; @blevins2021]. Moreover, this analysis was limited by the substantially smaller sample size (*n* = `r nrow(nacc_htn)`) than that of the NACC GWAS. Additionally, the diagnostic variables in NACC consisted of a mixture of patient self-reports, physician reporting, and use of HTN or DM medications. Future studies employing causal inference methods such as two-sample Mendelian randomization may be able to provide clearer evidence for or against the roles of HTN and DM in B-ASC and the mediation of B-ASC genetic risk by clinical risk factors and comorbidities.

An important limitation of our study is the relatively low sample size for a GWAS. NACC, the largest cohort used, had only `r nacc[, .N]` participants available for GWAS, and the total sample size of the Stage 3 mega-analysis with all four cohorts was `r act_mega[, .N]`. In contrast, a recent GWAS meta-analysis of AD exceed 90,000 participants @kunkle2019, and GWAS of other phenotypes have exceeded 1 million participants @lee2018. Nevertheless, smaller GWAS of other neuropathologic endophenotypes have identified risk loci [@katsumata2017], demonstrating that smaller sample sizes do not necessarily prevent discovery, especially for large effect-size phenomena. Despite sample size limitations, one suggestive locus on Chromosome 10 near *SORCS3* identified in NACC was validated in ROSMAP, and both this locus and a locus near *ELOVL4* were replicated in ADNI. Furthermore, nine of eleven loci suggestively associated with B-ASC in NACC had affect sizes in the same direction in ROSMAP, which suggests that these loci may be associated with B-ASC but suffer from suffer from regression towards the mean in the ROSMAP analysis. Furthermore, the use of downstream analyses can provide additional functional evidence for risk loci that do not meet the conservative Bonferroni-corrected significance thresholds.

Most genetic loci identified preliminarily as being associated with B-ASC in NACC were not validated in ROSMAP. This could be potentially explained via the significantly different demographic attributes and study designs between the two cohorts, as ROSMAP participants were older at death and had lower average levels of diagnosed B-ASC pathology (**Table 1**). Thus, there may be unaccounted-for confounding and differential selection bias between cohorts. For instance, ROS recruits from Catholic sisters and brothers, who are on average more highly educated than the general population. In contrast, the NACC participants are recruited from primarily clinic-based populations across the ADRCs. Differences in study design are also seen in the ADNI cohort, whose recruitment practices resemble those of ADRCs, and in ACT, whose participants are recruited from the aged dementia-free community in the Seattle, WA area.

Heterogeneity in neuropathological grading of B-ASC in each cohort may also contribute to impairing study power and limiting reproducibility between studies. In one study investigating the inter-rater reliability of cerebrovascular neuropathological diagnoses (i.e., how often neuropathologists agree on diagnosing a given slide), B-ASC had the second-lowest reliability (after microinfarcts), with reliability below 0.8 in five of twelve brain regions graded [@skrobot2016]. This relatively low inter-rater diagnostic reliability under near-ideal circumstances ((different trained experts reviewing the exact same stained slides) indicates there may be substantial measurement variance in B-ASC between neuropathologists both within and between study centers. Furthermore, B-ASC was graded in different brain regions in different centers and studies. As examples, B-ASC is graded globally in NACC, with no particular brain region specified, but is graded based on findings in the basal ganglia in ROSMAP @chou2013.

\newpage

# Funding Acknowledgements

**The author(s) disclosed receipt of the following financial support for the research, authorship, and/or publication of this article:** the University of Kentucky Center for Clinical and Translational Science TL-1 Fellowship [grant number TLITR001997]; the National Center for Advancing Translational Sciences [grant number UL1TR001998]; and the Dean of the College of Medicine, University of Kentucky. The content is solely the responsibility of the authors and does not necessarily represent the official views of the NIH or the University of Kentucky.

Genotyping was performed by Alzheimer's Disease Genetics Consortium (ADGC) [grant numbers U01 AG032984, RC2AG036528].

We thank the study participants and staff of the Rush Alzheimer's Disease Center. The ROSMAP studies are supported by NIA grants **xxxx**.

Data collection and sharing for this project was funded by the Alzheimer's Disease Neuroimaging Initiative (ADNI) (National Institutes of Health Grant U01 AG024904) and DOD ADNI (Department of Defense award number W81XWH-12-2-0012). ADNI is funded by the National Institute on Aging, the National Institute of Biomedical Imaging and Bioengineering, and through generous contributions from the following: AbbVie, Alzheimer's Association; Alzheimer's Drug Discovery Foundation; Araclon Biotech; BioClinica, Inc.; Biogen; Bristol-Myers Squibb Company; CereSpir, Inc.; Cogstate; Eisai Inc.; Elan Pharmaceuticals, Inc.; Eli Lilly and Company; EuroImmun; F. Hoffmann-La Roche Ltd and its affiliated company Genentech, Inc.; Fujirebio; GE Healthcare; IXICO Ltd.; Janssen Alzheimer Immunotherapy Research & Development, LLC.; Johnson & Johnson Pharmaceutical Research & Development LLC.; Lumosity; Lundbeck; Merck & Co., Inc.; Meso Scale Diagnostics, LLC.; NeuroRx Research; Neurotrack Technologies; Novartis Pharmaceuticals Corporation; Pfizer Inc.; Piramal Imaging; Servier; Takeda Pharmaceutical Company; and Transition Therapeutics. The Canadian Institutes of Health Research is providing funds to support ADNI clinical sites in Canada. Private sector contributions are facilitated by the Foundation for the National Institutes of Health ([www.fnih.org](https://fnih.org/)). The grantee organization is the Northern California Institute for Research and Education, and the study is coordinated by the Alzheimer's Therapeutic Research Institute at the University of Southern California. ADNI data are disseminated by the Laboratory for Neuro Imaging at the University of Southern California.

The NACC database is funded by NIA/NIH Grant U01 AG016976. NACC data are contributed by the NIA-funded ADRCs: P30 AG019610 (PI Eric Reiman, MD), P30 AG013846 (PI Neil Kowall, MD), P50 AG008702 (PI Scott Small, MD), P50 AG025688 (PI Allan Levey, MD, PhD), P50 AG047266 (PI Todd Golde, MD, PhD), P30 AG010133 (PI Andrew Saykin, PsyD), P50 AG005146 (PI Marilyn Albert, PhD), P50 AG005134 (PI Bradley Hyman, MD, PhD), P50 AG016574 (PI Ronald Petersen, MD, PhD), P50 AG005138 (PI Mary Sano, PhD), P30 AG008051 (PI Thomas Wisniewski, MD), P30 AG013854 (PI Robert Vassar, PhD), P30 AG008017 (PI Jeffrey Kaye, MD), P30 AG010161 (PI David Bennett, MD), P50 AG047366 (PI Victor Henderson, MD, MS), P30 AG010129 (PI Charles DeCarli, MD), P50 AG016573 (PI Frank LaFerla, PhD), P50 AG005131 (PI James Brewer, MD, PhD), P50 AG023501 (PI Bruce Miller, MD), P30 AG035982 (PI Russell Swerdlow, MD), P30 AG028383 (PI Linda Van Eldik, PhD), P30 AG053760 (PI Henry Paulson, MD, PhD), P30 AG010124 (PI John Trojanowski, MD, PhD), P50 AG005133 (PI Oscar Lopez, MD), P50 AG005142 (PI Helena Chui, MD), P30 AG012300 (PI Roger Rosenberg, MD), P30 AG049638 (PI Suzanne Craft, PhD), P50 AG005136 (PI Thomas Grabowski, MD), P50 AG033514 (PI Sanjay Asthana, MD, FRCP), P50 AG005681 (PI John Morris, MD), P50 AG047270 (PI Stephen Strittmatter, MD, PhD).

# Author Contributions

Lincoln M.P. Shade, Yuriko Katsumata, Peter T. Nelson, and David W. Fardo conceived of and carried out the research and drafted the manuscript. Timothy J. Hohman, Julie A. Schneider, Andrew J. Saykin, Shubabrata Mukherjee, Kevin L. Boehme, and John S.K. Kauwe provided data. Shubhabrata Mukherjee assisted with and provided consultation for ACT analyses. Kwangsik Nho and Andrew J. Saykin assisted with and provided consultation for ADNI analyses.

# Disclosures

Each author will need to add their financial discloscures and conflicts of interest before submission. If there are none, we will state: "The Authors declare that there is no conflict of interest." If there are some, those authors will indicate them, and then we will state: "All other authors declare that they have no conflict of interest."

# Supplementary Information

Supplementary materials are available online.

All code used for this study are available at <https://github.com/lincoln-shade/arter>.

\newpage

# References

::: {#refs}
:::

\newpage

# Figure Legends

**Figure 1.** Overall study design. First, quality control (QC) was performed on all neuropathological and genotype data sets used (see Methods). GWAS were performed across three stages. In Stage 1, GWAS was performed on NACC participants (n = 3382). In Stage 2, all variants with *p* \< $1\times10^{-5}$ (k = 14) were analyzed separately in ROS-MAP (n = 1187), and then NACC and ROS-MAP participant data were merged for mega-analysis (n = 4569). In Stage 3, top variants from Stage 2 mega-analysis with *p* \< $1\times10^{-6}$ (k = 2) were analyzed in two replication cohorts, ADNI (n = 47) and ACT (n = 512), and then NACC, ROS-MAP, ADNI, and ACT data were merged for mega-analysis. Downstream functional analyses, consisting of colocalization and gene-based analyses, were then performed on Stage 3 mega-analysis results. Mediation analyses for HTN and DM were also performed using a subset of NACC participants with clinical data available (n = 1726).

**Figure 2.** Stage 2 mega-analysis results. **A)** Manhattan plot of Stage 2 GWAS mega-analysis of NACC and ROSMAP. The horizontal dashed line in each plot is the suggestive threshold (*p* \< $1\times10^{-5}$) while the solid line is the threshold for genome-wide significance (*p* \< $5\times10^{-8}$). **B & C)** Regional plots of rs2603462 and rs7902929 +/- 1000 kb, respectively.Â 

# Tables

```{r table-1, include=TRUE}
# make table_1 object
table_1 <- autofit(table_1) %>% 
  padding(padding = 2, part = "all") %>% 
  add_header_lines("Table 1: Cohorts Used in Study") %>% 
  add_footer_lines("Key: SD, standard deviation; Min, minimum; Max, maximum. Age of Death variable is integer for NACC, ADNI, and ACT but continuous in ROSMAP.")

table_1
```

fdafdfasd

```{r table-2, include=TRUE}
table_2 %>% 
  autofit() %>% 
  padding(padding = 2, part = "all") %>% 
  add_header_lines("Table 2: Stage 1 GWAS Results") %>% 
  add_footer_lines("Key: SNP, single nucleotide polymorphism; CHR, Chromosome; BP, base pair; Gene, closest protein-coding gene; A1/A2, effect/non-effect allele; OR, odds ratio; 95% CI, 95% confidence interval. A1 set so that NACC OR >= 1.")

```

fdafdfasdf

```{r table-3, include = TRUE}
table_3_ncol <- ncol(table_3_data)

table_3 <- flextable(table_3_data)

table_3 %>%
  autofit() %>%
  padding(padding = 2, part = "all") %>%
  set_header_labels(nacc_rosmap_or_95ci = "OR [95% CI]", 
                    adni_or_95ci = "OR [95% CI]",
                    act_or_95ci = "OR [95% CI]",
                    P_ADNI = "P", 
                    P_ACT = "P") %>% 
  add_header_row(values = c(" ", "Stage 2 Mega-Analysis", "ADNI", "ACT"),
                 colwidths = c(4, 2, 2, 2)) %>% 
  add_header_lines(values = c("Table 3: Stage 2 Mega-Analysis and Replication Results")) %>% 
  add_footer_lines("Key: SNP, single nucleotide polymorphism; CHR, Chromosome; BP, base pair; Gene, closest protein-coding gene; A1/A2, effect/non-effect allele; OR, odds ratio; 95% CI, 95% confidence interval. A1 set so that Stage 2 mega-analysis OR >= 1.")

```

fdsafdsaf

```{r table-4, include=TRUE}
table_4 <- flextable(table_4_data)

table_4 %>% 
  autofit() %>% 
  padding(padding = 2, part = "all") %>% 
  add_header_lines(values = c("Table 4: Stage 3 Mega-Analysis Results")) %>% 
  add_footer_lines("Key: SNP, single nucleotide polymorphism; CHR, Chromosome; BP, base pair; Gene, closest protein-coding gene; A1/A2, effect/non-effect allele; OR, odds ratio; 95% CI, 95% confidence interval. A1 set so that OR >= 1.")
```

fdsafadsfdsa

```{r table-5, include=TRUE}
setorder(act_mega_coloc, CHR, -PPH4)
act_mega_coloc[, Tissue := stri_replace_first_fixed(Tissue, "_", ": ")]
act_mega_coloc[, Tissue := stri_replace_all_fixed(Tissue, "_", " ")]
table_5 <- act_mega_coloc[PPH4 > 50, .(SNP, CHR, Gene, Tissue, PPH4)] %>% 
  .[, Gene := c("NSD1", "ELOVL4", "AC058791.1", "AC058791.1", rep("WASF3", 3), "WASF3*", "FFAR3")] %>% 
  setnames("Gene", "Phenotype") %>% 
  flextable() %>%
  add_header_lines("Table 5: Expression Quantitative Trait Loci (eQTL) Colocalizing with B-ASC Risk") %>% 
  add_footer_lines("Key: SNP, single-nucleotide polymorphism; CHR, Chromosome; PPH$, posterior probability of colocalization") %>% 
  add_footer_lines("*Denotes sQTL phenotype")

table_5
```
